### PR TITLE
[core][state] Ignore tasks w/o status updates (driver generated tasks) by default from GCS

### LINKED ISSUE #31438: [CI] `linux://python/ray/tests:test_state_api` is failing/flaky on master.
- eb096f7a2f1e6ddaf930e64d760db26d424971e0 FAILED [Buildkite :python: (Large)](https://buildkite.com/ray-project/oss-ci-build-branch/builds/1647#01853c9f-7be2-45fe-bca4-d2928b5c8c8d)

....
Generated from flaky test tracker. Please do not edit the signature in this section.
DataCaseName-linux://python/ray/tests:test_state_api-END
....

The time it started to be flakey is around https://github.com/ray-project/ray/commit/eb096f7a2f1e6ddaf930e64d760db26d424971e0

### PR DESCRIPTION
**Pull request by rickyyx**
Signed-off-by: rickyyx <rickyx@anyscale.com>

## Why are these changes needed?
With profiling events (`submit_task`) collected on the driver, we will have entries of `TaskEvent` from the driver being reported to GCS. These events don't have any task spec info associated with them (only profile events with task id and job id).  When querying events in state API, the driver task event will get accounted for sometimes when truncation happens. This PR does a source-side filtering of driver events with `exclude_driver_task` to True to filter out driver tasks. 

## Related issue number
Closes https://github.com/ray-project/ray/issues/31438

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: src/ray/protobuf/gcs_service.proto [hunk 1]
```diff
@@ -624,6 +624,8 @@ message GetTaskEventsRequest {
   // If set, the exact `limit` TaskEvents returned do not have any ordering or selection
   // guarantee.
   optional int64 limit = 3;
+  // True if only task events with non-empty task state changes should be returned.
```
Comment (rkooo567) [inline src/ray/protobuf/gcs_service.proto:627]: Hmm this flag seems a bit confusing to me. Maybe it should be exclude_driver_task?

Comment (rickyyx) [reply src/ray/protobuf/gcs_service.proto:627]: I am ok with having `driver` mentioned in the flag - although it's a bit implicit where a driver task will generate task events w/o any status updates but only profiling events.

**Review by rkooo567 — APPROVED** [review]

**Comment by rkooo567** [comment]
cc @wuisawesome for the proto change

**Comment by rkooo567** [comment]
@rickyyx maybe you can ping alex in person... lmk when this is ready to be merged

**Review by scv119 — APPROVED** [review]

**Pull request merged**